XenAdmin/Commands/ShutDownHostCommand.cs

```diff
@@ -143,21 +145,44 @@ protected override string ConfirmationDialogText
 
                     if (h.Connection.ResolveAll(h.resident_VMs).Exists(v => v.HciWarnBeforeShutdown()))
                         hciHosts.Add(h);
+
+                    if (Helpers.HostIsMaster(h) && h.Connection.Cache.HostCount > 1)
+                        poolMasters.Add(h);
                 }
 
+                StringBuilder sb = new StringBuilder();
+                string firstWarning;
+
+
                 if (hciHosts.Count > 0)
-                    return hciHosts.Count == 1
-                        ? string.Format(Messages.CONFIRM_SHUTDOWN_HCI_WARN_SERVER, hciHosts[0].Name())
-                        : string.Format(Messages.CONFIRM_SHUTDOWN_HCI_WARN_SERVERS, string.Join("\n", hciHosts.Select(h => h.Name())));
-
-                if (hasRunningVMs)
-                    return hosts.Count == 1
-                        ? string.Format(Messages.CONFIRM_SHUTDOWN_SERVER, hosts[0].Name())
-                        : Messages.CONFIRM_SHUTDOWN_SERVERS;
-
-                return hosts.Count == 1
-                    ? string.Format(Messages.CONFIRM_SHUTDOWN_SERVER_NO_VMS, hosts[0].Name())
-                    : Messages.CONFIRM_SHUTDOWN_SERVERS_NO_VMS;
+                    firstWarning = hciHosts.Count == 1
+                        ? firstWarning = string.Format(Messages.CONFIRM_SHUTDOWN_HCI_WARN_SERVER, hciHosts[0].Name())

```

Comment:
"firstWarning = " assignments in line 159 and 160 are not needed the value is assigned in line 158